gtk4/src/dialog.rs

```diff
@@ -49,9 +67,16 @@ impl<O: IsA<Dialog>> DialogExtManual for O {
     fn get_response_for_widget<P: IsA<Widget>>(&self, widget: &P) -> ResponseType {
         unsafe {
             from_glib(ffi::gtk_dialog_get_response_for_widget(
-                self.as_ref().to_glib_none().0,
+                AsRef::<Dialog>::as_ref(self).to_glib_none().0,
                 widget.as_ref().to_glib_none().0,
             ))
         }
     }
+
+    fn run<F: Fn(&Self, ResponseType) + 'static>(&self, f: F) {
+        self.connect_response(move |s, response_type| {
+            f(s, response_type);
```

Comment:
You want to disconnect the signal here and close the dialog

Reply:
To do that I have to store it inside a `Arc<RefCell<>>` I guess...

Reply:
Just a `RefCell<Option<_>>` is enough